pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -287,6 +287,27 @@ func (p *Provisioner) reconcileCluster(cluster *v3.Cluster, create bool) (*v3.Cl
 		apiEndpoint, serviceAccountToken, caCert, err = p.driverUpdate(cluster, *spec)
 	} else {
 		logrus.Infof("Updating cluster [%s]", cluster.Name)
+
+		updating := false
+		for i := 0; i < 4; i++ {
+			if cluster, err = p.Clusters.Get(cluster.Name, metav1.GetOptions{}); err != nil {
```

Comment:
we should first try to perform an update, if it fails and is a conflict error, only then Get object